### PR TITLE
migrate to null safety

### LINKED ISSUE #15: Migrate to null safety
https://dart.dev/null-safety/migration-guide
https://flutter.dev/docs/null-safety

### PR DESCRIPTION
**Pull request by britannio**
Closes https://github.com/mcrovero/flutter_interactive_keyboard/issues/15

### CODE REVIEW

File: lib/src/keyboard_manager.dart [hunk 1]
```diff
@@ -57,21 +60,19 @@ class KeyboardManagerWidgetState extends State<KeyboardManagerWidget> {
...
+      if (!oldKeyboardOpen && activePointer == null) {
+        if (widget.onKeyboardOpen != null) widget.onKeyboardOpen!();
       }
     } else {
       // Close notification if the keyobard closes while not dragging
-      if(oldKeyboardOpen && activePointer == null) {
-        if(widget.onKeyboardClose != null)
-          widget.onKeyboardClose();
+      if (oldKeyboardOpen && activePointer == null) {
+        if (widget.onKeyboardClose != null) widget.onKeyboardClose!();
```
Comment (david-shortman) [inline lib/src/keyboard_manager.dart]: i see both `methodName?.call()` and `methodName!()` used in this PR. is there a reason to use one over another?

File: lib/src/channel_manager.dart [hunk 2]
```diff
@@ -19,13 +19,11 @@ class ChannelManager {
...
-    return await _channel.invokeMethod('expand');
-  }
+  static Future<bool> expand() =>
+      _channel.invokeMethod<bool>('expand').then((x) => x ?? false);
 
-  static Future<bool> fling(double velocity) async {
-    return await _channel.invokeMethod('fling', velocity);
-  }
+  static Future<bool> fling(double velocity) =>
+      _channel.invokeMethod<bool>('fling', velocity).then((x) => x ?? false);
```
Comment (david-shortman) [inline lib/src/channel_manager.dart:26]: i'm a little out-of-date on dart language updates. Like `Observables` used in JS, I guess `Future`s are now returned by synchronous methods?

makes sense to me.

**Review by david-shortman — COMMENTED** [review]
Since there's no automated testing, I'll have to pull up the example locally later to verify functionality is preserved.

Thanks for the PR!

File: lib/src/keyboard_manager.dart (hunk 1 above)
Comment (britannio) [reply lib/src/keyboard_manager.dart]: I've now removed all usages of `methodName!()` to minimise the use of the `!` cast. Incorrectly using the cast causes runtime exceptions like the one fixed by https://github.com/FirebaseExtended/flutterfire/pull/5656

File: lib/src/channel_manager.dart (hunk 2 above)
Comment (britannio) [reply lib/src/channel_manager.dart:26]: This has always been possible in Dart and it's slightly more concise for one-liners like this. 
`_channel.invokeMethod<bool>` returns a `Future<bool?>` and the `.then((x) ...` maps the output of `invokeMethod` to a non-nullable bool to match the return type of `fling`

**Review by david-shortman — APPROVED** [review]
Looks good to me locally. The package updates also appear appropriate.
Thanks again!

**Pull request merged**